src/Classes/Utils/CanvasUtils.ts

```diff
@@ -0,0 +1,348 @@
+        const start: ICoordinates = {
+            x: -radius,
+            y: radius / 1.66,
+        };
+
+        const side: ISide = {
+            count: {
+                denominator: 2,
+                numerator: 3,
+            },
+            length: radius * 2,
+        };
+
+        CanvasUtils.drawGenericPolygonShape(context, start, side);
+    }
+
+    private static drawPolygonShape(context: CanvasRenderingContext2D, radius: number, sides: number) {
+        const start: ICoordinates = {
+            x: -radius / (sides / 3.5),
+            y: -radius / (2.66 / 3.5),
+        };
+        const side: ISide = {
+            count: {
+                denominator: 1,
+                numerator: sides,
+            },
+            length: radius * 2.66 / (sides / 3),
+        };
+
+        CanvasUtils.drawGenericPolygonShape(context, start, side);
+    }
+
+    private static drawStarShape(context: CanvasRenderingContext2D, radius: number, sides: number) {
+        const start: ICoordinates = {
+            x: -radius * 2 / (sides / 4),
+            y: -radius / (2 * 2.66 / 3.5),
+        };
+        const side: ISide = {
+            count: {
+                denominator: 2,
+                numerator: sides,
+            },
+            length: radius * 2 * 2.66 / (sides / 3),
+        };
+
+        CanvasUtils.drawGenericPolygonShape(context, start, side);
+    }
+
+    private static drawLineShape(context: CanvasRenderingContext2D, length: number, stroke: IStroke) {
+        context.moveTo(0, -length/2);
+        context.lineTo(0, length / 2);
+        context.strokeStyle = stroke.color;
+        context.lineWidth = stroke.width;
+        context.stroke();
+    }
+
+    private static drawCircleShape(context: CanvasRenderingContext2D, radius: number, center: ICoordinates) {
+        context.arc(center.x, center.y, radius, 0, Math.PI * 2, false);
+    }
+
+    private static drawSquareShape(context: CanvasRenderingContext2D, side: number) {
+        context.rect(-side, -side, side * 2, side * 2);
+    }
+
+    private static drawHeartShape(context: CanvasRenderingContext2D, radius: number) {
+        const x = -radius / 2;
+        const y = -radius / 2;
+
+        context.moveTo(x, y + radius / 4);
+        context.quadraticCurveTo(x, y, x + radius / 4, y);
+        context.quadraticCurveTo(x + radius / 2, y, x + radius / 2, y + radius / 4);
+        context.quadraticCurveTo(x + radius / 2, y, x + radius * 3 / 4, y);
+        context.quadraticCurveTo(x + radius, y, x + radius, y + radius / 4);
+        context.quadraticCurveTo(x + radius, y + radius / 2, x + radius * 3 / 4, y + radius * 3 / 4);
+        context.lineTo(x + radius / 2, y + radius);
+        context.lineTo(x + radius / 4, y + radius * 3 / 4);
+        context.quadraticCurveTo(x, y + radius / 2, x, y + radius / 4);
+    }
+
+    private static drawTextShape(context: CanvasRenderingContext2D, text: string, pos: ICoordinates, fill: boolean) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected call-signature: 'drawTextShape' to have a typedef](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5171185)